Component/LinearSolver/Direct/src/sofa/component/linearsolver/direct/SparseCholeskySolver.inl

```diff
@@ -100,8 +140,78 @@ void SparseCholeskySolver<TMatrix,TVector>::invert(Matrix& M)
     A.nz = -1;							// # of entries in triplet matrix, -1 for compressed-col
     cs_dropzeros( &A );
     tmp.resize(A.n);
-    S = cs_schol (&A, order) ;		/* ordering and symbolic analysis */
-    N = cs_chol (&A, S) ;		/* numeric Cholesky factorization */
+
+    switch (type_perm.getValue().getSelectedId() )
+    {
+        case 0:
+        default://None->identity
+            {
+                sofa::helper::ScopedAdvancedTimer factorization_permTimer("factorization_perm");
+                S = symbolic_Chol (&A) ;		/* ordering and symbolic analysis */
+                N = cs_chol (&A, S) ;		/* numeric Cholesky factorization */
+                break;
+            }
+        case 1://SuiteSparse
+            {
+                sofa::helper::ScopedAdvancedTimer factorization_permTimer("factorization_perm");
+                int order = -1;
+                S = cs_schol (&A, order) ;		/* ordering and symbolic analysis */
+                N = cs_chol (&A, S) ;		/* numeric Cholesky factorization */
+            }
+            break;
+        case 2://METIS
+            perm.resize(A.n);
+            iperm.resize(A.n);
+            { 
+                sofa::helper::ScopedAdvancedTimer factorization_permTimer("factorization_perm");
+                if(computePermutation)
+                {
+                    fill_reducing_perm(A , perm.data(), iperm.data() ); // compute the fill reducing permutation
+                    computePermutation = false;
```

Comment:
Same remark than earlier: you need to check if the structure changed to detect if you need to recompute the permutation.